### PR TITLE
Rate limiter

### PR DESCRIPTION
**Pull request by 9helix**
## Description

Added rate limiter for the API using the Upstash rate limit.
Upstash account and Redis database created on it **needed**. More info: https://upstash.com/docs/redis/overall/getstarted
Credentials need to be provided in `.env` like this:
UPSTASH_REDIS_REST_URL="[database link]"
UPSTASH_REDIS_REST_TOKEN="[database token]"

## Change summary

- Initialized limiter
- Set up error throwing when limit is exceeded

## Checklist

- [ ] I have performed a self-review of my code
- [x] I tested locally
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New package or package update introduced

### CODE REVIEW

**Comment by 9helix** [comment]
Now I've added the ability to enable the limiter (disabled by default) from the `.env` file using the variable `RATE_LIMITER_ENABLED`. Set it to either `true` or `false`.

File: src/server/api/trpc.ts [hunk 1]
```diff
@@ -87,6 +88,17 @@ export const createTRPCRouter = t.router;
  */
 export const publicProcedure = t.procedure;
 
+import { Ratelimit } from "@upstash/ratelimit";
+import { Redis } from "@upstash/redis";
```
Comment (IMilja) [inline src/server/api/trpc.ts:92]: Could we move the imports to the top of the file?

File: src/env.js [hunk 2]
```diff
@@ -44,6 +49,7 @@ export const env = createEnv({
     NEXTAUTH_SECRET: process.env.NEXTAUTH_SECRET,
     NEXTAUTH_URL: process.env.NEXTAUTH_URL,
     ADMIN_PASSWORD: process.env.ADMIN_PASSWORD,
+    RATE_LIMITER_ENABLED: process.env.RATE_LIMITER_ENABLED,
```
Comment (IMilja) [inline src/env.js:52]: Could you add the `RATE_LIMITER_ENABLED` variable to the example .env file?

**Review by IMilja — APPROVED** [review]
Looking good to me. I requested two small changes and after you implement them we can merge the changes :)

**Pull request merged**